src/app/vault/organization-delete.component.ts

```diff
@@ -0,0 +1,98 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+
+import { CipherService } from '../../services/cipher.service';
+
+import { ApiService } from 'jslib/abstractions/api.service';
+import { CollectionService } from 'jslib/abstractions/collection.service';
+import { CryptoService } from 'jslib/abstractions/crypto.service';
+import { I18nService } from 'jslib/abstractions/i18n.service';
+import { PlatformUtilsService } from 'jslib/abstractions/platformUtils.service';
+import { SearchService } from 'jslib/abstractions/search.service';
+import { UserService } from 'jslib/abstractions/user.service';
+
+import { Organization } from 'jslib/models/domain/organization';
+import { PasswordVerificationRequest } from 'jslib/models/request/passwordVerificationRequest';
+import { CollectionView } from 'jslib/models/view/collectionView';
+
+@Component({
+    selector: 'app-organization-delete',
+    templateUrl: 'organization-delete.component.html',
+})
+export class OrganizationDeleteComponent implements OnInit {
+    @Input() organizationId: string;
+    @Output() onDeletedOrganization = new EventEmitter<Organization>();
+
+    title: string;
+    organizationName: string;
+    organization: Organization;
+    masterPassword: string;
+    formPromise: Promise<any>;
+
+    constructor(private apiService: ApiService, protected i18nService: I18nService,
+        protected platformUtilsService: PlatformUtilsService, private cryptoService: CryptoService,
+        private searchService: SearchService, private userService: UserService,
+        private collectionService: CollectionService, private cipherService: CipherService) { }
+
+    async ngOnInit() {
+        await this.init();
+    }
+
+    async removeAllCiphersFromCollection(collection: CollectionView) {
+        const collectionCiphers = await this.searchService.searchCiphers(
+            null,
+            [c => c.collectionIds != null && c.collectionIds.indexOf(collection.id) > -1],
+            null
+        );
+
+        for (const cipher of collectionCiphers) {
+            await this.cipherService.unshare(cipher);
+        }
+    }
+
+    async removeAllCollections() {
+        const allCollections = await this.collectionService.getAllDecrypted();
+        const collections = allCollections
+            .filter(collection => collection.organizationId === this.organizationId);
+
+        for (const collection of collections) {
+            await this.removeAllCiphersFromCollection(collection);
+        }

```

Comment:
can't we make a Promise.all() / race() or something that make parallelisation?

Reply:
I don't like to use `Promise.all()` to parallelise code that make HTTP queries because web browsers have a limit of parallel HTTP queries and this limit may differ between browsers and OS.

I encountered a lot of problems doing that a few years ago (i.e rejected requests or browser idling a few seconds between some requests), since I prefer to make them sequential.

But maybe the Promise.all has been improved since (it was not yet implemented in all browser at this time). Are you aware of changes on this behavior?

Reply:
Promise.all if not supported will be handled by babel so no worries. I don't know in which app, but we have an npm package to be able to provide the number of promises we want to run in //. Maybe in contacts.